### ISSUE TITLE
Support Guava immutable collections with `@Value.Modifiable`

### ISSUE THREAD

**Issue by rgabbard-bbn** [immutables/immutables#644]
After #643 is fixed, fields which are Guava immutable collections will be omitted from modifiable companion objects (produced using `@Value.Modifiable`).  Long-term, it would be nice for the companion object to internally have a mutable version of the relevant field.

For example,

```java
@Value.Immutable
@Value.Modifiable
abstract class Foo {
  public abstract ImmutableList<String> bar();
}
```

could generate

```java
final class ModifiableFoo extends Foo {
  private List<String> bar; // should probably be an ArrayList

  ....
  @Override
  public final List<String> bar() {
    return bar;
  }

...

  public ModifiableFoo addBar(String element) {
    String nonNullElement = Preconditions.checkNotNull(element, "bar element");
    bar.add(nonNullElement);
    return this;
  }
```

**Comment by elucash** [comment]
thank you for the feature request! I see a problem with returning `List<String>` if `ImmutableList<String>` was declared: contravariant overrides do not work in java, only the other way around. The workaround could be to store and modify mutable collection internally, but return immutable snapshot if someone is asking by calling getter on modifiable object (which is forced to return `ImmutableList` by abstract value type). This semantic seems whacky but modifiable object are super-awkward constructs anyway )

**Comment by elucash** [comment]
Released in 2.5.5, thank you!
